### PR TITLE
[google-cloud-cpp] use features to stop package growth

### PR DESCRIPTION
**Pull request by coryan**
We have plans to grow `google-cloud-cpp`, but we think the build times
are large enough already. This creates "features" to at least stop the
package growth as new features are added. The "default-features"
reflects the current features in the package.

- #### What does your PR fix?  

N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

No change.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes.

### CODE REVIEW

File: ports/google-cloud-cpp/portfile.cmake [hunk 1]
```diff
@@ -12,11 +12,24 @@ vcpkg_from_github(
...
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        bigtable GOOGLE_CLOUD_CPP_ENABLE_BIGTABLE
+        bigquery GOOGLE_CLOUD_CPP_ENABLE_BIGQUERY
+        firestore GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE
+        iam GOOGLE_CLOUD_CPP_ENABLE_IAM
+        pubsub GOOGLE_CLOUD_CPP_ENABLE_PUBSUB
+        spanner GOOGLE_CLOUD_CPP_ENABLE_SPANNER
+        experimental-storage-grpc GOOGLE_CLOUD_CPP_STORAGE_ENABLE_GRPC
```
Comment (dbolduc) [inline ports/google-cloud-cpp/portfile.cmake:24]: should there be a regular `storage` feature in here?

File: ports/google-cloud-cpp/portfile.cmake [hunk 2]
```diff
@@ -12,11 +12,25 @@ vcpkg_from_github(
...
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        bigquery GOOGLE_CLOUD_CPP_ENABLE_BIGQUERY
+        bigtable GOOGLE_CLOUD_CPP_ENABLE_BIGTABLE
+        firestore GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE
+        iam GOOGLE_CLOUD_CPP_ENABLE_IAM
+        pubsub GOOGLE_CLOUD_CPP_ENABLE_PUBSUB
+        spanner GOOGLE_CLOUD_CPP_ENABLE_SPANNER
+        storage GOOGLE_CLOUD_CPP_ENABLE_STORAGE
+        experimental-storage-grpc GOOGLE_CLOUD_CPP_STORAGE_ENABLE_GRPC
```
Comment (NancyLi1013) [inline ports/google-cloud-cpp/portfile.cmake:25]: Seems you missed some features here. `common`, `experimental-logging`, `googleapis`, `grpc-utils` are not listed, but these features are added in vcpkg.json file.

Comment (coryan) [reply ports/google-cloud-cpp/portfile.cmake:27]: Good points. I removed `common` and `grpc-utils`, these are infrastructure libraries automatically compiled if any of the other features are turned on.  I also fixed the support for `experimental-logging`.

File: ports/google-cloud-cpp/vcpkg.json [hunk 3]
```diff
@@ -20,5 +21,139 @@
...
+  ],
+  "features": {
+    "bigquery": {
+      "description": "The Google Cloud BigQuery C++ client library",
+      "dependencies": [
+        "abseil",
+        {
+          "name": "google-cloud-cpp",
+          "features": [
+            "common",
```
Comment (NancyLi1013) [inline ports/google-cloud-cpp/vcpkg.json:41]:
```suggestion
```
This should not exist here since it seems not a feature for `google-cloud-cpp`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Have you tested these features in this PR? @coryan

**Comment by coryan** [comment]
> Have you tested these features in this PR? @coryan

Evidently: not enough.

**Comment by coryan** [comment]
Closing for now, I think I need to make changes to the package CMake files to make this work well in all cases.

**Pull request closed without merging**